### ISSUE TITLE
rtc.getAlarmFlag(Alarm2) output true recurring when it should not

### ISSUE THREAD

**Issue by agfaps** [Erriez/ErriezDS3231#2]
Dear @Erriez , thank you for your library.
I have issue with alarm2. When I check for Alarm2 occurence in a loop, there is activation at a time when it should not activate.
This is the output from my code:
`19:08:45.724 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`19:48:45.753 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`21:10:15.649 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`00:21:05.032 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`01:42:35.648 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`03:01:41.943 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`03:25:46.864 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`03:29:18.890 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`04:50:56.583 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`05:16:45.625 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`
`10:37:56.892 -> Alarm2 ACTIVE: rtc.getAlarmFlag(Alarm2) == true`

after each check that rtc.getAlarmFlag(Alarm2) == true, I disable the alarm:
`rtc.clearAlarmFlag(Alarm2);`
 Can you give some hint about this issue that I am facing?
 Thank you.

**Comment by Erriez** [comment]
Hi @agfaps. Thanks for using my library. 

Function `getAlarmFlag(Alarm2)` returns the status of alarm flag 2 from the control/status register. When this alarm flag 2 bit is set, there is a time match with the alarm 2 registers and `getAlarmFlag(Alarm2` returns true.
It looks like the alarm 2 registers `A2M2`...`A2M4` (configurable via function `setAlarm2()`) minimum interval is one minute and maximum value is date, but cannot set to disabled. Only interrupt alarms 1 and 2 can be enabled or disabled, but alarm flags are always active.

Does this answer your question?

**Comment by agfaps** [comment]
Hi @Erriez , thank you for your reply.
In my code, I did not use interrupt to save pins. I use polling by poll alarm2 flag. I end up by storing the hour and minute of alarm on eeprom and when alarm2 flag is set, read hour and minute from ds3231 and compare this value with hour and minute stored in eeprom. If the value matched, then alarm2 flag occurence are valid. I don't know if this is the right method or not.
Previously I thought that alarm2 flag will only be active when there is match condition between A2M2 ... A2M4 with RTC registers. I also set the condition of alarm2 using setAlarm2.
Thank you.
